torchgen/model.py

```diff
@@ -581,7 +603,7 @@ def from_yaml(
             )
         elif not structured and structured_delegate is None:
             dispatch[DispatchKey.CompositeImplicitAutograd] = BackendMetadata(
-                cpp.name(func), structured=False
+                cpp.name(func), structured=False, cpp_namespace="at::native"
```

Comment:
Actually, the logic above says that "all kernels should live in some {foo}::native" namespace, but it's hard to enforce that invariant because you're obligated to write out the full cpp namespace every place that you construct a `BackendMetadata` object.

I think it would be less error prone if you had the `BackendMetadata` only take in the first part of the namespace (`at`), and have the __init__ function take care of appending a `::native` to it. And maybe add a `DEFAULT_NAMESPACE = "at"` global somewhere in the codegen that all of the call-sites have to use, to make it clear that it should be consistent everywhere.

Reply:
this one too, I think?

Reply:
Yeah we can't do this since lazy tensor codegen is having a namespace `torch::lazy`. We can't append `::native` to that.